CMakeLists.txt

```diff
@@ -42,6 +43,12 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "(Apple)?[Cc]lang")
         set(OT_CFLAGS -Werror)
     endif()
 
+    if(OT_COVERAGE)
+        list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_ENABLE_COVERAGE=1")
+        add_compile_options(-g -O0 --coverage)
```

Comment:
Is this specific to clang? What about gcc?

Reply:
I think this works for gcc as well. https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html#Instrumentation-Options

Reply:
Not clear to me if older versions of gcc (e.g. 4 and 5) support "--coverage", but I'm fine to leave this as is for now.